### PR TITLE
[SPARK-30934][ML][DOCS] Update ml-guide and ml-migration-guide for 3.0 release

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Update ml-guide and ml-migration-guide for 3.0.

### Why are the changes needed?
This is required for each release.

### Does this PR introduce any user-facing change?
Yes.
image

image

image

image

### How was this patch tested?
Manually build and check.

### CODE REVIEW

File: docs/ml-migration-guide.md [hunk 1]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
...
 {:.no_toc}
 
+**Deprecations**
+
+* [SPARK-11215](https://issues.apache.org/jira/browse/SPARK-11215):
+`labels` in `StringIndexer` is deprecated and will be removed in 3.1.0. Use `labelsArray` instead.
+* [SPARK-25758](https://issues.apache.org/jira/browse/SPARK-25758):
+`computeCost` in `BisectingKMeans` is deprecated and will be removed in future versions. Use `ClusteringEvaluator` instead.
+
+**Changes of behavior**
```
Comment (huaxingao) [inline docs/ml-migration-guide.md]: Do we need to include the newly added stuff in this section? For example, ```[SPARK-29960](https://issues.apache.org/jira/browse/SPARK-29960)
 `hammingLoss` support is addd to MulticlassClassificationEvaluator```? If we do, then there are a lot more to add.

**Review by srowen — APPROVED** [review]

**Comment by huaxingao** [comment]
Thanks @srowen for reviewing! Also cc @WeichenXu123 @zhengruifeng

File: docs/ml-guide.md [hunk 2]
```diff
@@ -87,31 +85,41 @@ To use MLlib in Python, you will need [NumPy](http://www.numpy.org) version 1.4
 [^1]: To learn more about the benefits and background of system optimised natives, you may wish to
     watch Sam Halliday's ScalaX talk on [High Performance Linear Algebra in Scala](http://fommil.github.io/scalax14/#/).
 
-# Highlights in 2.3
```
Comment (viirya) [inline docs/ml-guide.md]: This means we didn't remove it when releasing 2.4?

File: docs/ml-migration-guide.md [hunk 3]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
...
+* `org.apache.spark.mllib.regression.LassoWithSGD` which is deprecated in 2.0, is removed in 3.0, use `org.apache.spark.ml.regression.LinearRegression` with `elasticNetParam` = 1.0. Note the default `regParam` is 0.01 for `LassoWithSGD`, but is 0.0 for `LinearRegression`.
+* `org.apache.spark.mllib.regression.LinearRegressionWithSGD` which is deprecated in 2.0, is removed in 3.0, use `org.apache.spark.ml.regression.LinearRegression` or `LBFGS` instead.
+* `org.apache.spark.mllib.clustering.KMeans.getRuns` and `setRuns` which are deprecated in 2.1, is removed in 3.0, have no effect since Spark 2.0.0.
+* `org.apache.spark.ml.LinearSVCModel.setWeightCol` which is deprecated in 2.4, is removed in 3.0, is not intended for users.
+* From 3.0, `org.apache.spark.ml.classification.MultilayerPerceptronClassificationModel` extends `MultilayerPerceptronParams` to expose the training params. As a result, layers in `MultilayerPerceptronClassificationModel` has been changed from Array[Int] to IntArrayParam. Users should use `MultilayerPerceptronClassificationModel.getLayers` instead of `MultilayerPerceptronClassificationModel.layers` to retrieve the size of layers.
```
Comment (viirya) [inline docs/ml-migration-guide.md]: `layers` in `MultilayerPerceptronClassificationModel` has been changed from `Array[Int]` to `IntArrayParam`?

File: docs/ml-migration-guide.md [hunk 4]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
...
+* `org.apache.spark.ml.classification.GBTClassifier.numTrees`  which is deprecated in 2.4.5, is removed in 3.0, use `getNumTrees` instead.
+* `org.apache.spark.ml.clustering.KMeansModel.computeCost` which is deprecated in 2.4, is removed in 3.0, use `ClusteringEvaluator` instead.
+* `org.apache.spark.mllib.evaluation.MulticlassMetrics.precision` which is deprecated in 2.0, is removed in 3.0, use `accuracy` instead.
+* `org.apache.spark.mllib.evaluation.MulticlassMetrics.recall` which is deprecated in 2.0, is removed in 3.0, use `accuracy` instead.
+* `org.apache.spark.mllib.evaluation.MulticlassMetrics.fMeasure` which is deprecated in 2.0, is removed in 3.0, use `accuracy` instead.
```
Comment (viirya) [inline docs/ml-migration-guide.md]: I still saw `precision`, `recall` and `fMeasure` in `MulticlassMetrics`?

File: docs/ml-migration-guide.md [hunk 5]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
 
 * `OneHotEncoder` which is deprecated in 2.3, is removed in 3.0 and `OneHotEncoderEstimator` is now renamed to `OneHotEncoder`.
 * `org.apache.spark.ml.image.ImageSchema.readImages` which is deprecated in 2.3, is removed in 3.0, use `spark.read.format('image')` instead.
+* `org.apache.spark.mllib.clustering.KMeans.train` with param Int `runs` which is deprecated in 2.1, is removed in 3.0, use `train` method without `runs` instead.
```
Comment (viirya) [inline docs/ml-migration-guide.md]: Is it better to replace `... which is deprecated in 2.3, is removed in 3.0, use ...` with `... which is deprecated in 2.3, is removed in 3.0. Use ...`?

File: docs/ml-migration-guide.md [hunk 6]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
...
  `stringOrderType` param in `StringIndexer`, in case of equal frequency, the order of
  strings is undefined. Since Spark 3.0, the strings with equal frequency are further
  sorted by alphabet. And since Spark 3.0, `StringIndexer` supports encoding multiple
  columns.
+ * [SPARK-20604](https://issues.apache.org/jira/browse/SPARK-20604):
+ In prior to 3.0 releases, `Imputer` requires input column to be Double or Float. In 3.0, this
+ restriction is lifted so `Imputer` can handle all numeric types.
+* [SPARK-23469](https://issues.apache.org/jira/browse/SPARK-23469):
+In Spark 3.0, the `HashingTF` Transformer uses a corrected implementation of the murmur3 hash
+function to hash elements to vectors. `HashingTF` fits with Spark 3.0 will map elements to
```
Comment (viirya) [inline docs/ml-migration-guide.md]: Does ``` `HashingTF` fits ``` means ``` `HashingTF.fits` ``` here?

File: docs/ml-migration-guide.md [hunk 7]
```diff
@@ -33,16 +33,65 @@ Please refer [Migration Guide: SQL, Datasets and DataFrame](sql-migration-guide.
...
+different positions in vectors than in Spark 2. However, `HashingTF` created with Spark 2.x
+and loaded with Spark 3.0 will still use the previous hash function and will not change behavior.
+* [SPARK-28969](https://issues.apache.org/jira/browse/SPARK-28969):
+The `setClassifier` method in Pyspark's `OneVsRestModel` has been removed in 3.0 for parity with
```
Comment (viirya) [inline docs/ml-migration-guide.md]: Pyspark -> PySpark

**Review by viirya — COMMENTED** [review]
Thanks for the effort on updating the documents.

File: docs/ml-migration-guide.md (hunk 1 above)
Comment (zhengruifeng) [reply docs/ml-migration-guide.md]: I am netural on it, since it is not a big change.

File: docs/ml-migration-guide.md (hunk 3 above)
Comment (zhengruifeng) [reply docs/ml-migration-guide.md]: Yes, it was a `layers` field in `MultilayerPerceptronClassificationModel` before 3.0.
In 3.0, we make it extends `MultilayerPerceptronParams` and has param `layers`

File: docs/ml-guide.md [hunk 8]
```diff
@@ -87,31 +85,41 @@ To use MLlib in Python, you will need [NumPy](http://www.numpy.org) version 1.4
...
+([SPARK-23578](https://issues.apache.org/jira/browse/SPARK-23578)),
+([SPARK-11215](https://issues.apache.org/jira/browse/SPARK-11215)),
+([SPARK-29808](https://issues.apache.org/jira/browse/SPARK-29808)),
+([SPARK-22796](https://issues.apache.org/jira/browse/SPARK-22796)).
+* Support Tree-Based Feature Transformation was added
+([SPARK-13677](https://issues.apache.org/jira/browse/SPARK-13677)).
+* Two new evaluators `MultilabelClassificationEvaluator` and `RankingEvaluator` were added
+([SPARK-16692](https://issues.apache.org/jira/browse/SPARK-16692)),
+([SPARK-28045](https://issues.apache.org/jira/browse/SPARK-28045)).
+* Sample weights support was added in `DecisionTreeClassifier/Regressor`, `RandomForestClassifier/Regressor`, `BisectingKMeans`, `KMeans` and `GaussianMixture`
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply docs/ml-guide.md]: [24102][24103] RegressionEvaluator and BinaryClassificationEvaluator also support weighting

File: docs/ml-migration-guide.md (hunk 3 above)
Comment (viirya) [reply docs/ml-migration-guide.md]: yea, I just mean to add quotes to layers, Array[Int] and IntArrayParam.

File: docs/ml-migration-guide.md (hunk 4 above)
Comment (huaxingao) [reply docs/ml-migration-guide.md]: I mean the class member variables. Will change to 
```
The member variable `precision` in `org.apache.spark.mllib.evaluation.MulticlassMetrics` which is deprecated in 2.0, is removed in 3.0. Use `accuracy` instead.
The member variable `recall` in `org.apache.spark.mllib.evaluation.MulticlassMetrics` which is deprecated in 2.0, is removed in 3.0. Use `accuracy` instead.
The member variable `fMeasure` in `org.apache.spark.mllib.evaluation.MulticlassMetrics` which is deprecated in 2.0, is removed in 3.0. Use `accuracy` instead.
```

File: docs/ml-migration-guide.md (hunk 5 above)
Comment (huaxingao) [reply docs/ml-migration-guide.md]: Will change to 
```
`org.apache.spark.mllib.clustering.KMeans.train` with param Int `runs` which is deprecated in 2.1, is removed in 3.0. Use `train` method without `runs` instead.
```
Is there a better way to say this? @srowen

File: docs/ml-migration-guide.md (hunk 6 above)
Comment (huaxingao) [reply docs/ml-migration-guide.md]: I will change to 
```  `HashingTF` in Spark 3.0 ```

File: docs/ml-migration-guide.md (hunk 7 above)
Comment (huaxingao) [reply docs/ml-migration-guide.md]: Will update

File: docs/ml-guide.md [hunk 9]
```diff
@@ -87,31 +85,41 @@ To use MLlib in Python, you will need [NumPy](http://www.numpy.org) version 1.4
...
+* Fit with validation set was added to Gradient Boosted Trees in Python
+([SPARK-24333](https://issues.apache.org/jira/browse/SPARK-24333)).
+* [`RobustScaler`](ml-features.html#robustscaler) transformer was added
+([SPARK-28399](https://issues.apache.org/jira/browse/SPARK-28399)).
+* [`Factorization Machines`](ml-classification-regression.html#factorization-machines) classifier and regressor were added
+([SPARK-29224](https://issues.apache.org/jira/browse/SPARK-29224)).
+* Complement Naive Bayes Classifier was added
+([SPARK-29942](https://issues.apache.org/jira/browse/SPARK-29942)).
+* ML function parity between Scala and Python
+([SPARK-28958](https://issues.apache.org/jira/browse/SPARK-28958)).
```
[the remark this answers is not in the document]
Comment (huaxingao) [reply docs/ml-guide.md]: Will add all these you have mentioned.

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 10]
```diff
@@ -23,7 +23,7 @@ license: |
 `ALTER TABLE` statement changes the schema or properties of a table.
 
 ### RENAME 
-`ALTER TABLE RENAME` statement changes the table name of an existing table in the database.
+`ALTER TABLE RENAME TO` statement changes the table name of an existing table in the database.
```
Comment (srowen) [inline docs/sql-ref-syntax-ddl-alter-table.md]: Are these changes meant to be in this PR? looks somewhat unrelated

Comment (huaxingao) [reply docs/sql-ref-syntax-ddl-alter-table.md]: Sorry, I didn't realize this went in. Will remove now.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master/3.0

**Comment by huaxingao** [comment]
Thank you so much! @srowen @viirya @zhengruifeng
